torch/lib/TH/generic/THTensorMath.c

```diff
@@ -2925,6 +2917,35 @@ LAB_IMPLEMENT_BASIC_FUNCTION(neg,-)
 LAB_IMPLEMENT_BASIC_FUNCTION(cinv, TH_MATH_NAME(1.0) / )
 
 
+void THTensor_(pow)(THTensor *r_, THTensor *t, real value)
+{
+  THTensor_(resizeAs)(r_, t);
+  if(value == 1){
+    THTensor_(copy)(r_, t);
+  }
+  else if(value == 2){
+    THTensor_(cmul)(r_, t, t);
+  }
+  else if(value == 3){
+    TH_TENSOR_APPLY2(real, t, real, r_, *r__data = *t_data * *t_data * *t_data;);
+  }
+  else if(value == 0.5){
+    THTensor_(sqrt)(r_, t);
+  }
+  else if(value == -0.5){
+    THTensor_(rsqrt)(r_, t);
+  }
+  else if(value == -1){
+    THTensor_(cinv)(r_, t);
+  }
+  else if(value == -2){
+    TH_TENSOR_APPLY2(real, t, real, r_, *r__data = TH_MATH_NAME(1.0) / (*t_data * *t_data););
```

Comment:
`TH_MATH_NAME(1.0)` seems like a typo to me.

Reply:
`TH_MATH_NAME` is a macro that appends a `f` if the data type is float, else it don't do anything. I added it there to avoid possible casts from 1.0 (double) to 1.0f (float) when real is float. Such a pattern has already been used in [`cinv`](https://github.com/pytorch/pytorch/blob/master/torch/lib/TH/generic/THTensorMath.c#2925), so I think this should be ok.